### PR TITLE
Added tool types and reafactored tools to use tool types

### PR DESCRIPTION
**Pull request by purelogiq**
**This PR is part 2 of 3 of a bigger PR split up. Please merge it second**
This PR adds the basic CRUD for tool types and also refactors tools to use tool types.
It also includes changes to the db seeds files and new unit tests for the tool types addition.
This addresses issues #100

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -715,13 +715,26 @@
   nick = Participant.create({ andrewid: 'nharper', phone_number: 1713435788, user: nick_user })
   Membership.create({ organization: dtd_org, participant: nick })
 
-  puts 'Old Stuff...'
- tool = Tool.create({ name: 'Hammer', barcode: 7, description: 'it\'s a hammer' })
+  # Uncomment the line below and comment the creation of tool types and tools (so there is no conflicts)
+  #   to get a bunch of tools seeded into the database
```
Comment (minigeek) [inline db/seeds.rb]: This is confusing, what do you mean here? This is for test, right?

Comment (purelogiq) [inline db/seeds.rb]: Yes, normally when in the development environment only a few tools are created. You can uncomment that line and comment the liens below to get a bunch of tools seeded for the development environment to test with.

**Comment by purelogiq** [comment]
There may be a few minor ui quirks (can't think of any) that may already be fixed in the upcoming tool waitlist PR.

File: db/seeds.rb (hunk 1 above)
Comment (minigeek) [inline db/seeds.rb]: I would say we should always generate tool types regardless of environment (which we do for all other types such as shift types, etc.). I think the only option should be between 5 random tools and all the tools.

**Comment by pkoenig10** [comment]
This pull request doesn't include some of the things that you demoed to me on 3/3/2016, but I'm assuming those will be included in a later pull request.

**Pull request merged**